src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Facets/EditionFacet.cs

```diff
@@ -0,0 +1,138 @@
+				// However, we are not validating the instances of the oldItem and newItem as they should be either same instance either Equals.
+				var srcIndex = srcNode.IndexOf(oldItem, 0);
+				return new ReplaceNode(srcNode, oldItem, newItem, srcIndex);
+			});
+			_source.Update(RichNotifyCollectionChangedEventArgs.Replace(oldItem, newItem, index));
+		}
+	}
+
+	public void Add(object? value)
+	{
+		if (_service is { } svc)
+		{
+			var index = _collection.Count;
+			var newItem = value;
+
+			svc.Update(srcNode =>
+			{
+				// The source collection might have been updated and not yet pushed to the current thread.
+				// So to avoid out-of-range exception, we have to make sure to get the effective index from the source.
+				var srcIndex = Math.Min(index, srcNode.Count);
+				return new AddNode(srcNode, newItem!, srcIndex);
+			});
+			_source.Update(RichNotifyCollectionChangedEventArgs.Add(newItem, index));
+		}
+	}
+
+	public void Insert(int index, object? value)
+	{
+		if (_service is { } svc)
+		{
+			var newItem = value;
+
+			svc.Update(srcNode =>
+			{
+				// The source collection might have been updated and not yet pushed to the current thread.
+				// So to avoid out-of-range exception, we have to make sure to get the effective index from the source.
+				var srcIndex = Math.Min(index, srcNode.Count);
+				return new AddNode(srcNode, newItem!, srcIndex);
+			});
+			_source.Update(RichNotifyCollectionChangedEventArgs.Add(newItem, index));
+		}
+	}
+
+	public void RemoveAt(int index)
+	{
+		if (_service is { } svc)
+		{
+			var oldItem = _collection[index];
+
+			svc.Update(srcNode =>
+			{
+				// The source collection might have been updated and not yet pushed to the current thread.
+				// So to avoid out-of-range exception, we have to make sure to get the effective index from the source.
+				// However, we are not validating the instances of the oldItem and newItem as they should be either same instance either Equals.
+				var srcIndex = srcNode.IndexOf(oldItem, 0);
+				return srcIndex < 0 ? srcNode : new RemoveNode(srcNode, oldItem, srcIndex);
+			});
+			_source.Update(RichNotifyCollectionChangedEventArgs.Remove(oldItem, index));
+		}
+	}
+
+	public bool Remove(object? value)
+	{
+		if (_service is { } svc)
+		{
+			var index = _collection.IndexOf(value!);
+			if (index < 0)
+			{
+				return false;
+			}
+
+			var oldItem = value;
+
+			svc.Update(srcNode =>
+			{
+				// The source collection might have been updated and not yet pushed to the current thread.
+				// So to avoid out-of-range exception, we have to make sure to get the effective index from the source.
+				// However, we are not validating the instances of the oldItem and newItem as they should be either same instance either Equals.
+				var srcIndex = srcNode.IndexOf(oldItem, 0);
+				return srcIndex < 0 ? srcNode : new RemoveNode(srcNode, oldItem, srcIndex);
+			});
```

Comment:
This block seems to be duplicated multiple times in this file, is this on purpose?

Reply:
Well the comment is the same each time indeed, but otherwise the way we get the `srcIndex` sometimes differs and the "Node" is diferent.